base_layer/wallet_ffi/src/lib.rs

```diff
@@ -4135,6 +4138,106 @@ pub unsafe extern "C" fn wallet_get_seed_words(wallet: *mut TariWallet, error_ou
     }
 }
 
+/// Gets the current emoji set
+///
+/// ## Arguments
+/// `()` - Does not take any arguments
+///
+/// ## Returns
+/// `*mut EmojiSet` - Pointer to the created EmojiSet.
+///
+/// # Safety
+/// The ```emoji_set_destroy``` function must be called when finished with a ByteVector to prevent a memory leak
+#[no_mangle]
+pub unsafe extern "C" fn get_emoji_set() -> *mut EmojiSet {
+    let current_emoji_vec = emoji_set();
+    let mut emoji_set: Vec<ByteVector> = Vec::new();
+    for emoji in current_emoji_vec {
+        let mut b = [0; 4]; // emojis are 4 bytes, unicode character
+        let emoji_char = ByteVector(emoji.encode_utf8(&mut b).as_bytes().to_vec());
+        emoji_set.push(emoji_char);
+    }
+    let result = EmojiSet(emoji_set);
+    Box::into_raw(Box::new(result))
+}
```

Comment:
Single allocation of `emoji_set` Vec

```suggestion
/// Gets the emoji set
///
/// ## Returns
/// `*mut EmojiSet` - Pointer to the created EmojiSet.
///
/// # Safety
/// The ```emoji_set_destroy``` function must be called when finished with a ByteVector to prevent a memory leak
#[no_mangle]
pub unsafe extern "C" fn get_emoji_set() -> *mut EmojiSet {
    let current_emoji_set = emoji_set();
    let mut emoji_set = Vec::with_capacity(current_emoji_set.len());
    for emoji in current_emoji_set {
        let mut b = [0; 4]; // emojis are 4 bytes, unicode character
        let emoji_char = ByteVector(emoji.encode_utf8(&mut b).as_bytes().to_vec());
        emoji_set.push(emoji_char);
    }
    let result = EmojiSet(emoji_set);
    Box::into_raw(Box::new(result))
}
```

Reply:
The operative line is : ` let mut emoji_set = Vec::with_capacity(current_emoji_set.len());`